Sources/KIF/Classes/KIFUIViewTestActor.m

```diff
@@ -385,6 +406,31 @@ - (void)swipeFromEdge:(UIRectEdge)edge
     [self.actor swipeFromEdge:edge];
 }
 
+- (void)activateCustomActionWithName:(NSString *)name;
+{
+    KIFUIObject *found = [[self usingCustomActionWithName:name] _predicateSearchWithRequiresMatch:YES mustBeTappable:NO];
+    
+    NSArray *actions = [found.element accessibilityCustomActions];
+    
+    [self runBlock:^KIFTestStepResult(NSError **error) {
+        for (UIAccessibilityCustomAction *action in actions) {
+            NSString *actionName = [action name];
+            if ([actionName isKindOfClass:[NSAttributedString class]]) {
+                actionName = [(NSAttributedString *)actionName string];
+            }
+
+            if ([actionName isEqualToString: name]) {
+                if([action activate]) {
```

Comment:
How do VoiceOver and other accessibility tools respond if a custom action returns false? Are there valid use cases where a custom action might want to return false and that be a valid test scenario rather than treating it as a test failure? We don't need to extend to support that right now, but it seemed worth some thought/discussion. 

Is it reasonable that a custom action might fail in some cases? Or is the expectation that custom actions should generally only be present on an element if they're expected to be able to be performed? Looking at the [Firefox iOS codebase](https://github.com/mozilla-mobile/firefox-ios/blob/d8be21cb8b7ec11087138f11166883870daa6571/firefox-ios/Client/Frontend/Browser/BrowserViewController/Views/BrowserViewController.swift#L726) as an example, it seems like they filter the actions to ones that are expected to be usable in the current context, and then always return true. That's just one example though.

Reply:
theres no official dogma on this particular question, but as we interpret the default behavior of UIKit and SwiftUI as guidance it would suggest that custom actions that are known to fail should be omitted entirely. 

To the person in front of the device, a failed action will play an additional "error" sound when activated. 

You're right, we should consider the case where an action is expected to occasionally fail and allow for testing that assumption.